Assemblies TOC (#15717)
dotnet/docs: docs/standard/assembly/unloadability.md

Starting with .NET Core 3.0, the ability to load and later unload a set of assemblies is supported. In .NET Framework, custom app domains were used for this purpose, but .NET Core only supports a single default app domain.

> .NET Core 3.0 and later versions support unloadability through <xref:System.Runtime.Loader.AssemblyLoadContext>. You can load a set of assemblies into a collectible `AssemblyLoadContext`, execute methods in them or just inspect them using reflection, and finally unload the `AssemblyLoadContext`. That unloads the assemblies loaded into the `AssemblyLoadContext`.

Reviewer: I think `that` was referring to the previously mentioned `AssemblyLoadContext`.

Author: I agree but I'm not sure what your point is?

Reviewer: I meant that the original text was making sense.

Author: Agreed, but it had too many "thats".

- Regular references held from outside of the collectible `AssemblyLoadContext` that are stored in a stack slot or a processor register (method locals, either explicitly created by the user code or implicitly by the just-in-time (JIT) compiler), a static variable, or a strong (pinning) GC handle, and transitively pointing to:
  - An assembly loaded into the collectible `AssemblyLoadContext`.
  - A type from such an assembly.
  - An instance of a type from such an assembly.
- Threads running code from an assembly loaded into the collectible `AssemblyLoadContext`.
- Instances of custom, non-collectible `AssemblyLoadContext` types created inside of the collectible `AssemblyLoadContext`.
- Pending <xref:System.Threading.RegisteredWaitHandle> instances with callbacks set to methods in the custom `AssemblyLoadContext`.

> [!TIP]
> Object references that are stored in stack slots or processor registers and that could prevent unloading of an `AssemblyLoadContext` can occur in the following situations:
>
> - When function call results are passed directly to another function, even though there is no user-created local variable.

> > - When a reference to an object is available at any point in a method, and the JIT compiler keeps the reference for as long as it wants in the current function.

Reviewer 2: Suggest omit "for as long as it wants in the current function" or:

```suggestion
> - When the JIT compiler keeps a reference to an object that was available at some point in a method.
```

plugin load /path/to/libsosplugin.so
```

> Let's debug an example program that has problems with unloading. Source code is included below. When you run it under WinDbg, the program breaks into the debugger right after attempting to check for the unload success. You can then start looking for the culprits.

Reviewer 2: It would be ideal to get rid of first-person altogether, but as it appears to be pervasive in the doc it's probably not worth the effort.
